### PR TITLE
[API Compatiblity] mixed type in static image mode（roll、flatten、logical_*）

### PR DESCRIPTION
**Pull request by zhengshengning**
### PR Category

Operator Mechanism

### PR Types

New features

### Description

合并下面三个pr内容。
**PR1**：https://github.com/PaddlePaddle/Paddle/pull/74858
支持静态图模式下 PyObject 可以是混合类型（例如：[int,Value,Value] ）；
1. 增加 PyObject_CheckIRVectorOfValueOrLong。
2. 增加 CastPyArg2VectorOfValueOrLong。

静态图下可支持下面参数类型：
```
self.axis = [
            0,
            paddle.to_tensor([1], 'int64'),
            paddle.to_tensor([2], 'int64'),
        ]
```

**PR2**：https://github.com/PaddlePaddle/Paddle/pull/74907
下沉paddle.roll、paddle.flatten 和 paddle.Tensor.flatten 到C++层，支持Torch的传参方法。
**流程：**
1. 在ops.yaml增加 python_api 配置，增加映射关系和PreProcess函数；
2. 在_paddle_docs.py增加英文文档信息，add_doc_and_signature绑定；
3. 删掉原有实现，使用from _C_ops import xxx 代替;
4. 在arg_pre_process.h与arg_pre_process.cpp中实现PreProcess函数；
5. 删掉报错的老静态图（Old IR）的单测；
6. 增加单测
**其他：**
修复python_c_gen代码生成的bug。
避免nn.functional.sigmoid 引起的单测报错

**PR3**：https://github.com/PaddlePaddle/Paddle/pull/74733
Fix compatibility in paddle.logical_and, paddle.logical_not, paddle.logical_or, paddle.logical_xor

pcard-67164

### CODE REVIEW

**Review by wanghuancoder — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Comment by swgu98** [comment]
3个pr因冲突合并提交，此前分别都通过了Coverage
